Engine/AlgorithmManager.cs

```diff
@@ -201,6 +202,16 @@ public void Run(AlgorithmNodePacket job, IAlgorithm algorithm, IDataFeed feed, I
                 }
             }
 
+            // add internal feeds for currencies added during initialization
+            var addedSecurities = algorithm.Portfolio.CashBook.EnsureCurrencyDataFeeds(algorithm.Securities, algorithm.SubscriptionManager,
+                MarketHoursDatabase.FromDataFolder(), SymbolPropertiesDatabase.FromDataFolder(),
+                DefaultBrokerageModel.DefaultMarketMap, SecurityChanges.None);
+            foreach (var security in addedSecurities)
```

Comment:
For posterity:

We can do it this way, but it just feels like the wrong way to get these subscriptions into the data feed. My original suggestion was added of these securities is a setup handler concern and not an algorithm manager concern. Another thought was to use the universe system to ensure these subscriptions exist within a universe -- but that can be a later refactor.